srml/support/src/storage/generator/double_map.rs

```diff
@@ -0,0 +1,170 @@
+use sr_std::prelude::*;
+use codec::{Codec, Encode, EncodeAppend};
+use crate::{storage::{self, unhashed, hashed::StorageHasher}, rstd::borrow::Borrow};
+
+/// Generator for `StorageDoubleMap` used by `decl_storage`
+pub trait StorageDoubleMap<K1: Encode, K2: Encode, V: Codec> {
+	/// The type that get/take returns.
+	type Query;
+
+	/// Hasher for the first key.
+	type Hasher1: StorageHasher;
+
+	/// Hasher for the second key.
+	type Hasher2: StorageHasher;
+
+	/// Get the prefix key in storage.
+	fn key1_prefix() -> &'static [u8];
```

Comment:
nit: I think it's fine to call this `prefix` -- it prefixes both keys.

Reply:
actually it only prefixes the first key as final key is:
hasher1(key1_prefix ++ key1) ++ hasher2(key2)